_posts/2020-08-07-g2p.md

```diff
@@ -392,40 +406,215 @@ You can also use abbreviations like this in the G2P studio by writing them in th
 
 ### Advanced: use in Python
 
-...more to come...
+#### Installation
+
+We recommend installing `g2p` using [pip](https://en.wikipedia.org/wiki/Pip_(package_manager)):
+
+```python
+pip install g2p
+```
+
+If you are going to be creating your own mapping or editing `g2p` in any way, you must first fork and clone `g2p`:
+
+```bash
+git clone https://github.com/YourGitHubUsername/g2p.git
+```
+
+and then do an *editable* pip installation:
+
+```bash
+cd g2p && pip install -e .
+```
+
+#### Usage
+
+Using `g2p` within Python can be done, either programatically using the `make_g2p` function:
+
+```python
+>>> from g2p import make_g2p
+>>> transducer = make_g2p('dan', 'eng-arpabet')
+>>> transducer('hej').output_string
+'HH EH Y'
+```
+
+#### Command Line Interface
+
+You can also use `g2p` from the command line. The basic command for conversions is:
+
+```bash
+g2p convert <input_text> <in_lang> <out_lang>
+```
+
+So in practice:
+
+```bash
+$ g2p convert hej dan eng-arpabet
+HH EH YY
+```
 
 ## Advanced: contributing your rules to the main `g2p` library
 
-You've written some cool rules and you want to contribute, that's awesome! 
+You've written some cool rules and you want to contribute, that's awesome! There are lots of benefits to contributing your mapping to `g2p`. First of all, once your mapping is accepted, you'll have it available and live on [G2P Studio](https://g2p-studio.herokuapp.com). Second, once the next version of `g2p` is release with your mapping, it will be automatically built in to the [Convertextract](#convertextract) library. Third, if your mapping is between a language's writing system and the IPA, you can also get ReadAlongs support for your language.
+
+So, you write your mapping once, and you get three things for free (G2P studio, convertextract and readalongs). Here's how:
+
+1. Fork g2p
+2. Add a folder for your language using the appropriate [ISO 639.3 code](https://en.wikipedia.org/wiki/List_of_ISO_639-3_codes) to `g2p/mappings/langs`
+3. Add a config.yaml file as described [here](#mapping-configuration)
+4. Add your mapping in that same folder (`g2p/mappings/langs/<yourlang>`)
+5. If your mapping is for an IPA mapping, you can optionally [generate the mapping](#generate-your-mapping-between-your-languages-ipa-and-english-ipa) between your language and English IPA.
+6. Run `g2p update` to add your mapping to `g2p`
+7. Add some test data to `g2p/tests/public/data`. 
+8. Submit your changes by creating a [pull request](https://github.com/roedoejet/g2p/compare)
 
-...more to come...
+That's it! Either myself, or somebody else will review the changes, and you will get credit for those mappings and be added to the [list of contributors](https://github.com/roedoejet/g2p/graphs/contributors)
```

Comment:
```suggestion
Finally, either myself, or somebody else will review the changes, and you will get credit for those mappings and be added to the [list of contributors](https://github.com/roedoejet/g2p/graphs/contributors)
```

"That's it" might sound like this is a trivial process, when I know firsthand that it can be quite involved!